enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,973 @@
+
+* We assume that there already exist tests that check management cluster update and guest cluster update.
+
+* Nice to have: test that runs the management cluster outside of AWS (either bare metal or in another cloud) and
+  the guest in AWS to ensure that the control-plane parts for AWS EBS CSI driver work outside of AWS.
+
+<!--
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+- What additional testing is necessary to support managed OpenShift service-based offerings?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+-->
+
+### Graduation Criteria
+
+<!--
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+  - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+  - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+**If this is a user facing change requiring new or updated documentation in [openshift-docs](https://github.com/openshift/openshift-docs/),
+please be sure to include in the graduation criteria.**
+
+**Examples**: These are generalized examples to consider, in addition
+to the aforementioned [maturity levels][maturity-levels].
+-->
+
+#### Dev Preview -> Tech Preview
+
+N/A, going GA directly (!)
+
+#### Tech Preview -> GA
+
+* E2E CI jobs introduced above are stable.
+* Both the management cluster upgrade and the guest cluster upgrade CI jobs are stable.
+
+#### Removing a deprecated feature
+
+N/A?
+
+### Upgrade / Downgrade Strategy
+
+In general, the upgrade will be very similar to standalone cluster, just the new operators will not be started by
+CVO, but by control-plane-operator.
```

Comment:
It may be worth mentioning here whether clusters created with previous versions of HyperShift (4.10, 4.11) can be migrated to 4.12. 
I would expect not, but it would be good to explicitly state it.

Reply:
Good idea, added.